app/forms/declaration/officeOfExit/OfficeOfExit.scala

```diff
@@ -30,6 +30,15 @@ object OfficeOfExit {
     case _                                => OfficeOfExit(None, Some(AllowedUKOfficeOfExitAnswers.no))
   }
 
+  def from(officeOfExitInsideUK: OfficeOfExitInsideUK, officeOfExit: OfficeOfExit): OfficeOfExit =
```

Comment:
Would it not be simpler to always call this method from OfficeOfExitController and then to
- always copy over the 'isUkOfficeOfExit' value
- update the 'officeId' only if it exists on the form, otherwise keep the existing value (if any)

e.g.  
`OfficeOfExit(officeOfExitInsideUK.officeId.else(officeOfExit.officeId), officeOfExitInsideUK. isUkOfficeOfExit)`

Reply:
Maybe rename `officeOfExit` to `existingValue` and change it to be Option[OfficeOfExit] as you might not have an existing value.  Would make the logic in the controller above easier.

Reply:
The first comment won't work, as we incur the risk of populating the officeOfExit (outside UK) with an office from the UK.
I have updated the logic, hopefully more easier to read. This was actually my first approach with Leszek, but for some stupid reason ended up making it more complex than it needed to be.